app/models/miq_server.rb

```diff
@@ -369,6 +369,13 @@ def monitor_loop
       _log.info "Server Monitoring Complete - Timings: #{timings.inspect}" unless timings[:total_time] < server_log_timings_threshold
       sleep monitor_poll
     end
+  rescue Interrupt => e
+    _log.info("Recieved #{e.message} signal, killing server")
+    self.class.kill
+    exit 1
```

Comment:
I think it's really strange for a regular method on a model to exit 1, however that being said, this is no normal method.  We really need to rip out these "MiqServer" classes into standalone classes.